src/CSET/operators/ageofair.py

```diff
@@ -0,0 +1,311 @@
+                    if cyclic:
+                        if (
+                            x <= -1
+                        ):  # as for example -0.3 would still be in domain, but x_arr.shape-0.3 would result in index error
+                            x = x_arr.shape[3] + x  # wrap back around dateline
+                        elif x >= x_arr.shape[3]:
+                            x = x_arr.shape[3] - x
+                    else:
+                        if x < 0 or x >= x_arr.shape[3]:
+                            ageofair_local[leadtime, lat_pnt] = n * dt
+                            outside_lam = True
+
+                    if y < 0 or y >= x_arr.shape[2]:
+                        ageofair_local[leadtime, lat_pnt] = n * dt
+                        outside_lam = True
+
+    # Save 3d array containing age of air
+    np.save(tmpdir + "/aoa_frag_" + str(lon_pnt).zfill(4) + ".npy", ageofair_local)
+
+
+def compute_ageofair(
+    cubelist, plev, includevertical=True, cyclic=False, tmpdir="/tmp/", multicore=None
+):
+    """
+    Compute back trajectories for a particular timeslice of a forecast.
+
+    Arguments:
+        cubelist: an iris.CubeList object, containing 4 cubes (u, v, w component of wind, with geopotential
+                  height. All cubes 4D, structured time, pressure, latitude, longitude. Assumed correct
+                  ordering of arrays. Expects data to already be regridded onto a standard 0p5 degree grid.
+        plev: pressure level  to evaluate the back trajectory on (etc 700 or 70000) depending on hPa or Pa.
+        includevertical: If True, use vertical velocity vector to compute back trajectories. If False, then
+                         assume flow laminar (set vertical velocity to zero).
+        tmpdir: Location of a temporary directory that can be used to write intermediate files, as used
+                when multiprocessing.
+        multicore: If set as None, then run on a single thread. Otherwise, split over number of multicores.
+    """
+    # Filter cubelist and pull out required cubes (sometimes multiple with same name, need to check)
+    for cube in cubelist:
+        if cube.standard_name == "x_wind":
+            x_wind = cube
+        elif cube.standard_name == "y_wind":
+            y_wind = cube
+        elif cube.standard_name == "upward_air_velocity":
+            z_wind = cube
+        elif cube.standard_name == "geopotential_height":
+            geopot = cube
+
+    # Check that all cubes are of same dimension (some might differ slightly, extra y point often)
+    if not x_wind.shape == y_wind.shape == z_wind.shape == geopot.shape:
+        quit("Cubes not same dimlen! Quitting...")
+
+    # Get time units and assign for later
+    if str(x_wind.coord("time").units) == "hours since 1970-01-01 00:00:00":
+        timeunit = "hour"
+    else:
+        quit("Unsupported time base! Quitting...")
+
+    # Make data non-lazy to speed up code.
+    print("Making data non-lazy...")
```

Comment:
[Logging](https://docs.python.org/3/library/logging.html) should be used rather than print inside of a library. For most cases `logging.info("Message")` is appropriate, but if an error is occouring use `logging.error("Message")`, or if it is really detailed debugging information use `logging.debug("Message")`.